digest/digest.go

```diff
@@ -123,6 +123,20 @@ func (d Digest) Algorithm() Algorithm {
 	return Algorithm(d[:d.sepIndex()])
 }
 
+// Verifier returns a writer object that can be used to verify a stream of
+// content against the digest. If the digest is invalid, an error will be
+// returned.
+func (d Digest) Verifier() (Verifier, error) {
+	if err := d.Validate(); err != nil {
```

Comment:
if panic is acceptable then arguably this should be

```
if err := d.Validate(); err != nil {
	panic(err)
}
```

But that tends to freak people out. Just trying to avoid creating need for unnecessary error checks when we don't want to support that use case in the first place.